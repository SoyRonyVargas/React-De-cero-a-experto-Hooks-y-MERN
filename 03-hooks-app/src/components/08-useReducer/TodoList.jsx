import TodoListItem from './TodoListItem'
import React from 'react'

const TodoList = ({ todos, handleDelete }) => (
    <ul className='list-group list-group-flush col'>
        {
            todos.map((todo, index) => (
                <TodoListItem
                    key={index}
                    handleDelete={handleDelete}
                    index={index}
                    {...todo}
                />
            ))
        }
    </ul>
)

export default TodoList