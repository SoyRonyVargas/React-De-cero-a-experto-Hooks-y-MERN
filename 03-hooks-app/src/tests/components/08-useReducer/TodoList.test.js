import TodoList from '../../../components/08-useReducer/TodoList'
import { demoTodos } from '../../fixtures'
import { shallow } from 'enzyme'

describe('pruebas al componente <TodoList/>', () => { 

    const payload = {
        handleDelete: jest.fn(),
        todos: demoTodos
    }
    
    const wrapper = shallow(
        <TodoList
            {...payload}
        />
    )

    test('deberia mostrarse correctamente', () => { 

        expect(wrapper).toMatchSnapshot()

    })

    test('deberia de tener dos <TodoListItem/>', () => { 

        const elements = wrapper.find("TodoListItem") 

        expect(elements.length).toBe(2)

        expect(wrapper.find("TodoListItem").at(0).prop("handleDelete") ).toEqual( expect.any(Function) )

    })

})