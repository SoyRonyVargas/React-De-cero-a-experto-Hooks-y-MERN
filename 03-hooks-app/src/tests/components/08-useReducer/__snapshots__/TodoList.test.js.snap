// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas al componente <TodoList/> deberia mostrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush col"
>
  <TodoListItem
    desc="Aprender React ⚛"
    handleDelete={[MockFunction]}
    id={0}
    index={0}
    key="0"
  />
  <TodoListItem
    desc="Comer 🍴"
    handleDelete={[MockFunction]}
    id={1}
    index={1}
    key="1"
  />
</ul>
`;
